### PR TITLE
Package all overrides into a struct

### PR DESCRIPTION
**Pull request by mbenmclean**
This change adds support for --set-file helm overrides. It also packages all of the helm overrides into a struct. This struct has fields for the --set args, --set-string args, --set-file args, and -f args. This struct gets populated in the buildCustomHelmOverrides method. It defines a HelmComponent struct in the helm package to hold these overrides.

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: pkg/bom/bom.go [hunk 1]
```diff
@@ -110,6 +110,7 @@ type KeyValue struct {
 	Key       string
 	Value     string
 	SetString bool
+	SetFile   bool
```
Comment (michael-cico) [inline pkg/bom/bom.go:113]: Maybe we should add some comments here as to what these mean...

Comment (michael-cico) [reply pkg/bom/bom.go:113]: i.e., the difference between SetFile and IsFile... we can do that as a separate pass though.

File: platform-operator/internal/helm/helmcli.go [hunk 2]
```diff
@@ -26,6 +26,14 @@ const ChartStatusFailed = "failed"
 // Package-level var and functions to allow overriding GetChartStatus for unit test purposes
 type ChartStatusFnType func(releaseName string, namespace string) (string, error)
 
+// HelmOverrides contains all of the overrides that gets passed to the helm cli runner
+type HelmOverrides struct {
+	SetOverrides       string   // for --set
```
Comment (michael-cico) [inline platform-operator/internal/helm/helmcli.go:31]: Yeah, I think comments like these would be good in `bom.KeyValue`, but perhaps next pass.

**Review by michael-cico — APPROVED** [review]

File: platform-operator/internal/helm/helmcli.go (hunk 2 above)
Comment (mbenmclean) [reply platform-operator/internal/helm/helmcli.go:31]: I can throw them in now it's not a big deal

**Review by michael-cico — APPROVED** [review]

**Pull request merged**